src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderRepository.cs

```diff
@@ -153,7 +154,7 @@ public Task PutAsync(List<ProvenBlockHeader> headers, HashHeightPair newTip)
             {
                 this.logger.LogTrace("(-)[BLOCKHASH_MISMATCH]");
 
-                throw new InvalidOperationException("Invalid newTip block hash.");
+                throw new ProvenBlockHeaderException("Invalid new tip hash. The new tip hash is the same a the most recent proven block header tip hash that was saved to disk.");
```

Comment:
Maybe make this text shorter
`Invalid new tip hash, tip hash has not changed`

Reply:
Ok.